workflow-service/src/test/java/com/redhat/parodos/workflow/registry/BeanWorkFlowRegistryImplTest.java

```diff
@@ -0,0 +1,166 @@
+						if (data.isEmpty()) {
+							return false;
+						}
+						return data.containsKey("foo_dependency");
+					}
+					return false;
+				}));
+		assertEquals(registry.getWorkFlowByName("test"), workFlow);
+	}
+
+	@Test
+	void TestRegistryWithValidWorkflowTasks() {
+
+		// given
+		Work work = Mockito.mock(Work.class);
+		SequentialFlow workFlow = SequentialFlow.Builder.aNewSequentialFlow().named("test").execute(work).build();
+
+		Mockito.when(this.beanFactory.getDependenciesForBean(Mockito.eq("test")))
+				.thenReturn(new String[] { "foo_dependency" });
+		AnnotatedTypeMetadata bean = Mockito.mock(AnnotatedTypeMetadata.class);
+		BeanDefinition beanDefinition = Mockito.mock(BeanDefinition.class);
+		Mockito.when(beanDefinition.getSource()).thenReturn(bean);
+		Mockito.when(this.beanFactory.getBeanDefinition(Mockito.any())).thenReturn(beanDefinition);
+
+		Mockito.when(this.beanFactory.getDependenciesForBean(Mockito.eq("test-task")))
+				.thenReturn(new String[] { "task_dependency" });
+		WorkFlowTask testTask = Mockito.mock(WorkFlowTask.class);
+
+		// when
+		BeanWorkFlowRegistryImpl registry = new BeanWorkFlowRegistryImpl(this.beanFactory,
+				this.workFlowDefinitionService, new HashMap<String, WorkFlowTask>() {
+					{
+						put("test-task", null);
+					}
+				}, new HashMap<String, WorkFlow>() {
+					{
+						put("test", workFlow);
+					}
+				});
+		registry.postInit();
+
+		// then
+		Mockito.verify(this.beanFactory, Mockito.times(2)).getDependenciesForBean(Mockito.any());
+		Mockito.verify(this.workFlowDefinitionService, Mockito.times(1)).save(Mockito.eq("test"), Mockito.eq("test"),
+				Mockito.eq(WorkFlowType.ASSESSMENT), Mockito.argThat(argument -> {
+					if (argument instanceof HashMap) {
+						HashMap<?, ?> data = ((HashMap<?, ?>) argument);
+						if (data.isEmpty()) {
+							return false;
+						}
+						return data.containsKey("foo_dependency");
+					}
+					return false;
+				}));
+
+		Mockito.verify(this.workFlowDefinitionService, Mockito.times(1)).saveWorkFlowChecker(Mockito.eq("test-task"),
+				Mockito.eq("task_dependency"), Mockito.any());
+	}
+
+	@Test
+	void TestThatBeanIsAnnotatedTypeMetadata() {
+
+		// given
+		Work work = Mockito.mock(Work.class);
+		SequentialFlow workFlow = SequentialFlow.Builder.aNewSequentialFlow().named("test").execute(work).build();
+
+		Mockito.when(this.beanFactory.getDependenciesForBean(Mockito.eq("test")))
+				.thenReturn(new String[] { "foo_dependency" });
+
+		BeanDefinition beanDefinition = Mockito.mock(BeanDefinition.class);
+		Mockito.when(this.beanFactory.getBeanDefinition(Mockito.any())).thenReturn(beanDefinition);
+
+		// when
+		BeanWorkFlowRegistryImpl registry = new BeanWorkFlowRegistryImpl(this.beanFactory,
+				this.workFlowDefinitionService, new HashMap<String, WorkFlowTask>(), new HashMap<String, WorkFlow>() {
+					{
+						put("test", workFlow);
+					}
+				});
+		Exception exception = assertThrows(RuntimeException.class, () -> registry.postInit());
```

Comment:
I'd like to suggest to use method reference `registry::postInit` instead of lambda expression `() -> registry.postInit()`